File: java/complete/myretro/src/main/java/com/apress/myretro/retro/RetroBoard.java
package com.apress.myretro.retro;

import com.apress.myretro.user.User;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class RetroBoard {

    private UUID id;

    private String name;

    private Map<String,User> assignedUsers = new HashMap<>();

    private Map<UUID,Card> cards = new HashMap<>();

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private final LocalDateTime createdAt = LocalDateTime.now();

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime modifiedAt = LocalDateTime.now();
}

File: kotlin/complete/myretro/src/main/kotlin/com/apress/myretro/retro/RetroBoard.kt
package com.apress.myretro.retro

import com.apress.myretro.user.User
import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime
import java.util.HashMap
import java.util.UUID

data class RetroBoard(
    var id: UUID?,
    var name: String?,
    var assignedUsers: HashMap<String, User>? = hashMapOf(),
    var cards: HashMap<UUID, Card>? = hashMapOf(),
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    var modifiedAt:LocalDateTime? = LocalDateTime.now()
) {
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    var createdAt:LocalDateTime = LocalDateTime.now()

    constructor():this(UUID.randomUUID(),null, hashMapOf(), hashMapOf(),null)
}